apex/provider/resource_mobility_targets_test.go

```diff
@@ -0,0 +1,177 @@
+					}
+
+					FunctionMocker = Mock(helper.GetMobilityTarget).Return(nil, &http.Response{Body: io.NopCloser(strings.NewReader("")), StatusCode: 400}, fmt.Errorf("Mock error")).Build()
+				},
+				Config:      ProviderConfig + mobilityTargetResourceUpdateConfig,
+				ExpectError: regexp.MustCompile(`.*Could not read Mobility target*.`),
+			},
+			{
+				// Update failure
+				PreConfig: func() {
+					if FunctionMocker != nil {
+						FunctionMocker.UnPatch()
+					}
+					FunctionMocker = Mock(helper.UpdateMobilityTarget).Return(nil, nil, fmt.Errorf("Mock error")).Build()
+				},
+				Config:      ProviderConfig + mobilityTargetResourceUpdateConfig,
+				ExpectError: regexp.MustCompile(".*Error executing Update Mobility Target Job*"),
+			},
+
+			{
+				// Error while waiting for job after update
+				PreConfig: func() {
+					if FunctionMocker != nil {
+						FunctionMocker.UnPatch()
+					}
+
+					FunctionMocker = Mock(helper.WaitForJobToComplete).Return(nil, fmt.Errorf("Mock error")).Build()
+				},
+				Config:      ProviderConfig + mobilityTargetResourceUpdateConfig,
+				ExpectError: regexp.MustCompile(`.*Error getting resourceID*.`),
+			},
+		},
+	})
+}
+
+func TestAccMobilityTargetsResourceCreateError(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				// Create failure
+				PreConfig: func() {
+					if FunctionMocker != nil {
+						FunctionMocker.UnPatch()
+					}
+					FunctionMocker = Mock(helper.CreateMobilityTarget).Return(nil, nil, fmt.Errorf("Mock error")).Build()
+				},
+				Config:      ProviderConfig + mobilityTargetResourceConfig,
+				ExpectError: regexp.MustCompile(".*Could not create Mobility Target*"),
+			},
+			{
+				// Read failure after create
+				PreConfig: func() {
+					if FunctionMocker != nil {
+						FunctionMocker.UnPatch()
+					}
+
+					FunctionMocker = Mock(helper.GetMobilityTarget).Return(nil, &http.Response{Body: io.NopCloser(strings.NewReader("")), StatusCode: 400}, fmt.Errorf("Mock error")).Build()
+				},
+				Config:      ProviderConfig + mobilityTargetResourceConfig,
+				ExpectError: regexp.MustCompile(`.*Could not retrieve Mobility group*.`),
+			},
+			{
+				// Error while waiting for job after create
+				PreConfig: func() {
+					if FunctionMocker != nil {
+						FunctionMocker.UnPatch()
+					}
+
+					FunctionMocker = Mock(helper.WaitForJobToComplete).Return(nil, fmt.Errorf("Mock error")).Build()
+				},
+				Config:      ProviderConfig + mobilityTargetResourceConfig,
+				ExpectError: regexp.MustCompile(`.*Error getting resourceID*.`),
+			},
+		},
+	})
+}
+
+var mobilityTargetResourceConfig = `
+resource "apex_navigator_mobility_targets" "example" {
+	name                     = "create-mobility-target"
+	source_mobility_group_id = "POWERFLEX-ELMSIOENG10015__DATAMOBILITYGROUP__98cc3059-15ef-4634-a94d-66b3ab79ab8f"
```

Comment:
Should some of these be input variables so we could run the tests against an actual powerflex eventually

Reply:
Yes good point. Will add variables for these.